classes/Rijndael.php

```diff
@@ -119,11 +119,19 @@ public function decrypt($cipherText)
     protected function generateHmac($encrypted)
     {
         $macKey = $this->generateKeygenS2k('sha256', $this->_key, $this->_iv, 32);
-        return hash_hmac(
-            'sha256',
-            $this->_iv . MCRYPT_RIJNDAEL_128 . $encrypted,
-            $macKey
-        );
+        if(defined('MCRYPT_RIJNDAEL_128')) {
+            return hash_hmac(
+                'sha256',
+                $this->_iv . MCRYPT_RIJNDAEL_128 . $encrypted,
+                $macKey
+            );
+        }else{
+            return hash_hmac(
+                'sha256',
+                $this->_iv  . $encrypted,
```

Comment:
I think you need to rather hard-code name of the cipher otherwise old cookies will not work.